hoomd/md/TwoStepNPTMTK.cc

```diff
@@ -839,10 +876,90 @@ void TwoStepNPTMTK::advanceThermostat(unsigned int timestep)
     setIntegratorVariables(v);
     }
 
+void TwoStepNPTMTK::setCouple(const std::string& value)
+    {
+    bool twod = m_sysdef->getNDimensions()==2;
+    if (!(twod))
+        {
+        if (value == "none")
+            {
+            m_couple = couple_none;
+            }
+        else if (value == "xy")
+            {
+            m_couple = couple_xy;
+            }
+        else if (value == "xz")
+            {
+            m_couple = couple_xz;
+            }
+        else if (value == "yz")
+            {
+            m_couple = couple_yz;
+            }
+        else if (value == "xyz")
+            {
+            m_couple = couple_xyz;
+            }
+        }
+    // if sys_def is 2D, silently ignore any couplings that involve z
+    if (twod)
+        {
+        if (value == "none")
+            {
+            m_couple = couple_none;
+            }
+        else if (value == "xy")
+            {
+            m_couple = couple_xy;
+            }
+        else if (value == "xz")
+            {
+            m_couple = couple_none;
+            m_exec_conf->msg->warning() << "z coupling is ignored, since system is two dimensional. coupling is none." << endl;
+            }
+        else if (value == "yz")
+            {
+            m_couple = couple_none;
+            m_exec_conf->msg->warning() << "z coupling is ignored, since system is two dimensional. coupling is none." << endl;
+            }
+        else if (value == "xyz")
+            {
+            m_couple = couple_xy;
+            m_exec_conf->msg->warning() << "z coupling is ignored, since system is two dimensional. coupling is xy." << endl;
+            }
+        }
+    }
+ 
+std::string TwoStepNPTMTK::getCouple()
+    {
+    std::string couple;
+    if (m_couple == couple_none)
+        {
+        couple = "none";
+        }
+    else if (m_couple == couple_xy)
+        {
+        couple = "xy";
+        }
+    else if (m_couple == couple_xz)
+        {
+        couple = "xz";
+        }
+    else if (m_couple == couple_yz)
+        {
+        couple = "yz";
+        }
+    else if (m_couple == couple_xyz)
+        {
+        couple = "xyz";
+        }
+    return couple;
```

Comment:
This is a stylistic choice, but I would prefer to see a switch statement here https://www.tutorialspoint.com/cplusplus/cpp_switch_statement.htm